src/graph/SpatialMergePointsNode.cpp

```diff
@@ -67,13 +67,9 @@ void SpatialMergePointsNode::enqueueExecImpl()
 
 	// This could work lazily - merging only on demand
 	for (const auto& [field, data] : mergedData) {
-		data->resize(width, false, false);
-		size_t offset = 0;
 		for (const auto& input : pointInputs) {
-			size_t pointCount = input->getPointCount();
 			const auto toMergeData = input->getFieldData(field);
-			data->insertData(toMergeData->getReadPtr(MemLoc::Device), pointCount, offset);
-			offset += pointCount;
+			data->appendFrom(toMergeData);
 		}
 	}
```

Comment:
We should clear `data` before the second for-loop to delete data from the previous graph run.
As of now, it also merges points temporally.